components/ui/DropDown/DropDown.tsx

```diff
@@ -0,0 +1,81 @@
+import { cva, VariantProps } from "class-variance-authority";
+import { Menu, Transition } from "@headlessui/react";
+import { Fragment } from "react";
+import { ChevronDownIcon } from "@heroicons/react/20/solid";
+
+const dropDownStyles = cva("", {
+  variants: {
+    items: {
+      primary: "bg-blue-200",
+      secondary: "bg-green-200",
+    },
+  },
+  defaultVariants: {
+    items: "primary",
+  },
+});
+
+export interface MenuItemProps {
+  name: string;
+  items: string;
+}
+
+export const MenuItem = ({ name, items }: MenuItemProps) => {
+  return (
+    <Menu.Item>
+      {({ active }) => (
+        <button
+          className={`${
+            active ? dropDownStyles({ items }) : null
+          } group flex w-full items-center rounded-md px-2 py-2 text-sm`}
+        >
+          {active ? (
+            <div className="mr-2 h-5 w-5" aria-hidden="true" />
+          ) : (
+            <div className="mr-2 h-5 w-5" aria-hidden="true" />
+          )}
+          {name}
+        </button>
+      )}
+    </Menu.Item>
+  );
+};
+
+const MenuItems = ["option 1", "option 2", "option 3"];
+
+export interface DropDownProps extends VariantProps<typeof dropDownStyles> {}
+
+export const DropDown = ({ items, ...props }: DropDownProps) => {
+  return (
+    <div>
+      <Menu as="div" className="relative inline-block text-left">
+        <div>
+          <Menu.Button className="focus-visible:ring-white inline-flex w-full justify-center rounded-md bg-blue-400 px-4 py-2 text-sm font-medium text-white hover:bg-opacity-30 focus:outline-none focus-visible:ring-2 focus-visible:ring-opacity-75">
+            Click Me
+            <ChevronDownIcon
+              className="ml-2 -mr-1 h-5 w-5 text-brand-200 hover:text-brand-100"
+              aria-hidden="true"
+            />
+          </Menu.Button>
+        </div>
+        <Transition
+          as={Fragment}
+          enter="transition ease-out duration-100"
+          enterFrom="transform opacity-0 scale-95"
+          enterTo="transform opacity-100 scale-100"
+          leave="transition ease-in duration-75"
+          leaveFrom="transform opacity-100 scale-100"
+          leaveTo="transform opacity-0 scale-95"
+        >
+          <Menu.Items className="ring-black absolute right-0 mt-2 w-56 origin-top-right divide-y divide-gray-100 rounded-md bg-white shadow-lg ring-1 ring-opacity-5 focus:outline-none">
```

Comment:
could we extract out the className here into a `menuItemsContainerStyles` cva function?

Reply:
probably needs a dark mode style as well